memory.x

```diff
@@ -45,3 +45,25 @@ _stack_start = ORIGIN(RAM) + LENGTH(RAM);
 /* The location of the .text section can be overridden using the
    `_stext` symbol.  By default it will place after .vector_table */
 /* _stext = ORIGIN(FLASH) + 0x40c; */
+SECTIONS {
+  .itcm : ALIGN(8) {
+    *(.itcm .itcm.*);
+    . = ALIGN(8);
+    } > ITCM
+  .axisram : ALIGN(8) {
+    *(.axisram .axisram.*);
+    . = ALIGN(8);
+    } > AXISRAM
+  .sram1 (NOLOAD) : ALIGN(4) {
+    *(.sram1 .sram1.*);
+    . = ALIGN(4);
+    } > SRAM1
+  .sram2 (NOLOAD) : ALIGN(4) {
+    *(.sram2 .sram2.*);
+    . = ALIGN(4);
+    } > SRAM2
+  .sram3 (NOLOAD) : ALIGN(4) {
+    *(.sram3 .sram3.*);
+    . = ALIGN(4);
+    } > SRAM3
+} INSERT AFTER .bss;
```

Comment:
These are needed for any examples involving DMA, but I think we should be more selective about which ones we allow to be used. My suggestion would be:

* Add a comment explaining that these are used for examples
* Remove `.itcm`
* Replace `.sram1` and `.sram2` with a comment explaining that they are commonly used for the CM4 stack/heap, so should not be used in examples
* Add `.sram4`

That gives one RAM on each of the bus matrixes, which should be fine.

Reply:
Also, adding these invalidate the line L23 `/* Section definitions below this line are not used by stm32h7xx-hal */`